airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/AutoDisableConnectionActivityImpl.java

```diff
@@ -91,18 +91,22 @@ public AutoDisableConnectionOutput autoDisableFailingConnection(final AutoDisabl
         final Job firstJob = jobPersistence.getFirstReplicationJob(input.getConnectionId())
             .orElseThrow(() -> new Exception("Auto-Disable Connection should not have been attempted if no replication job has been run."));
         final int numDaysSinceFirstReplicationJob = getDaysSinceTimestamp(currTimestampInSeconds, firstJob.getCreatedAtInSecond());
+        final boolean firstReplicationOlderThanMaxDisableDays = numDaysSinceFirstReplicationJob >= maxDaysOfOnlyFailedJobs;
+        final boolean noPreviousSuccess = successTimestamp.isEmpty();
 
         // disable connection if only failed jobs in the past maxDaysOfOnlyFailedJobs days
-        if (numDaysSinceFirstReplicationJob >= maxDaysOfOnlyFailedJobs && successTimestamp.isEmpty()) {
+        if (firstReplicationOlderThanMaxDisableDays && noPreviousSuccess) {
           disableConnection(input.getConnectionId(), lastJob);
           return new AutoDisableConnectionOutput(true);
         }
 
-        // sending warning for disable connection if only failed jobs in the past
-        // maxDaysOfOnlyFailedJobsBeforeWarning days
-        if (numDaysSinceFirstReplicationJob >= maxDaysOfOnlyFailedJobsBeforeWarning &&
-            (successTimestamp.isEmpty()
-                || getDaysSinceTimestamp(currTimestampInSeconds, successTimestamp.get()) >= maxDaysOfOnlyFailedJobsBeforeWarning)) {
+        final boolean firstReplicationOlderThanMaxDisableWarningDays = numDaysSinceFirstReplicationJob >= maxDaysOfOnlyFailedJobsBeforeWarning;
+        final boolean successOlderThanPrevFailureByMaxWarningDays = // set to true if no previous success is found
+            noPreviousSuccess || getDaysSinceTimestamp(currTimestampInSeconds, successTimestamp.get()) >= maxDaysOfOnlyFailedJobsBeforeWarning;
+
+        // send warning if there are only failed jobs in the past maxDaysOfOnlyFailedJobsBeforeWarning days
+        // _except_ if a warning should have already been sent in the previous failure
```

Comment:
super nit: instead of `_except_ if a warning...`, go with `_unless_ a warning...`

Reply:
done